### PR TITLE
Switch from mamba to uv

### PR DESCRIPTION
**Pull request by chillenzer**
#5077 seems to have been insufficient.

### CODE REVIEW

**Review by psychocoderHPC — DISMISSED** [review]

**Comment by BrianMarre** [comment]
PyPIConGPU test fails, might have something to do with this setting https://github.com/ComputationalRadiationPhysics/picongpu/blob/89b9640670127075f753c4bc63ec92aa99f5b1d6/.gitlab-ci.yml#L184

**Comment by psychocoderHPC** [comment]
> PyPIConGPU test fails, might have something to do with this setting
> 
> https://github.com/ComputationalRadiationPhysics/picongpu/blob/89b9640670127075f753c4bc63ec92aa99f5b1d6/.gitlab-ci.yml#L184

@chillenzer I stopped the CI because this must be fixed

**Comment by chillenzer** [comment]
Thanks for pointing this out. As far as I can see it, the syntax `.*` is not supported by `uv` for selecting a Python version. Pushed an update. Btw, why aren't we on `3.12`?

**Comment by BrianMarre** [comment]
Because when we built the PyPIConGPU CI Python 3.12 wasn't out yet

**Comment by psychocoderHPC** [comment]
I stopped the CI again because of errors https://gitlab.com/hzdr/crp/picongpu/-/jobs/7785698774

**Comment by psychocoderHPC** [comment]
https://github.com/ComputationalRadiationPhysics/picongpu/blob/817b9ebbfec871bf76e8f374a7a974153aeb0663/share/ci/pypicongpu_generator.py#L327

is adding `.*` to the version number

**Comment by chillenzer** [comment]
Nasty. I have now also run `grep -ri "python" share` and `grep -r '\.\* share` with no further results.

File: share/ci/install/pypicongpu.sh [hunk 1]
```diff
@@ -27,20 +27,14 @@ cd $CI_PROJECT_DIR
 # use miniconda as python environment
```
Comment (PrometheusPi) [inline share/ci/install/pypicongpu.sh:27]: comment is now outdated

**Review by PrometheusPi — CHANGES_REQUESTED** [review]
minor comment correction

File: share/ci/install/pypicongpu.sh (hunk 1 above)
Comment (chillenzer) [reply share/ci/install/pypicongpu.sh:27]: Good catch. I'll remove it once the CI passed such that we can merge without another CI run.

**Comment by psychocoderHPC** [comment]
It looks like it is not as simple as it sounds to switch the package manager for python.
Let's evaluate if we go on with the switch or stay with micromambda. For now, it looks like IT solved the issues by updating their black list.

**Comment by chillenzer** [comment]
Apparently, IT has fixed the problem with `mamba` which makes this change unnecessary. Also, we have `snakemake` workflows which would make it particularly inconvenient to leave `mamba` completely. So, we'll close this PR for now.

PS: The failing CI probably just needs a renaming `pip3` -> `[uv ]pip` (not sure if we'd need/want `uv` in front there).

**Comment by BrianMarre** [comment]
For future reference, Simeon recommended to switch to tox to simplify the PyPIConGPU generator, since tox apparently provides a native way to fix software versions for CI tests.

**Pull request closed without merging**